### ISSUE TITLE
Model.get_config() returns a list instead of a dictionary.

### ISSUE THREAD

**Issue by SpirosChadoulos** [keras-team/keras#11179]
Hi,
In the documentation it is mentioned that model.get_config() returns a dictionary containing the configuration of the model, however when I use it it returns a list of dictionaries.

Configuration: Latest version of Keras & Tensorflow, Python 3.5, Linux Ubuntu 16.04.4 LTS (GNU/Linux 4.4.0-133-generic x86_64)

**Comment by gabrieldemarmiesse** [comment]
Can you give a small code snippet to reproduce? It seems strange to me that get_config returns a list.

**Comment by SpirosChadoulos** [comment]
I can't give you exact code to reproduce because it is part of a private project but I can tell you that I trained a neural network with 5 dense layers. I think that if you train a very simple neural network with dense layers and call the get_config() function you will receive a list of dictionaries.
`print(type(model.get_config()))`
Output: <class 'list'>

**Comment by gabrieldemarmiesse** [comment]
You can change the neural network, also the names of the variable. I'm asking because I can't reproduce this behavior. (changing the number of layers won't change the type of the config).
```python
from keras.applications import MobileNet
model = MobileNet(weights=None)
print(type(model.get_config()))
#<class 'dict'>
```
Without any script to reproduce, we have no chance of fixing a bug. Can you please help us on this?

**Comment by SpirosChadoulos** [comment]
I used this example from [https://towardsdatascience.com/how-to-build-a-neural-network-with-keras-e8faa33d0ae4](url) and I was able to reproduce the bug. 

```python
import numpy as np
from keras.utils import to_categorical
from keras import models
from keras import layers
from keras.datasets import imdb

(training_data, training_targets), (testing_data, testing_targets) = imdb.load_data(num_words=10000)

data = np.concatenate((training_data, testing_data), axis=0)

targets = np.concatenate((training_targets, testing_targets), axis=0)

def vectorize(sequences, dimension = 10000):
 results = np.zeros((len(sequences), dimension))
 for i, sequence in enumerate(sequences):
  results[i, sequence] = 1
 return results
 
data = vectorize(data)
targets = np.array(targets).astype("float32")

test_x = data[:10000]
test_y = targets[:10000]
train_x = data[10000:]
train_y = targets[10000:]

model = models.Sequential()

model.add(layers.Dense(50, activation = "relu", input_shape=(10000, )))

model.add(layers.Dropout(0.3, noise_shape=None, seed=None))
model.add(layers.Dense(50, activation = "relu"))
model.add(layers.Dropout(0.2, noise_shape=None, seed=None))
model.add(layers.Dense(50, activation = "relu"))

model.add(layers.Dense(1, activation = "sigmoid"))
model.summary()

model.compile(
 optimizer = "adam",
 loss = "binary_crossentropy",
 metrics = ["accuracy"]
)
results = model.fit(
 train_x, train_y,
 epochs= 2,
 batch_size = 500,
 validation_data = (test_x, test_y)
)
print(type(model.get_config()))
#<class 'list'>
```

**Comment by gabrieldemarmiesse** [comment]
I ran the script you gave and the result is `<class 'dict'>`. Can you do `pip freeze` in the command line and give me the result?

**Comment by SpirosChadoulos** [comment]
absl-py==0.2.2
aniso8601==3.0.2
astetik==1.9.5
astor==0.7.1
backcall==0.1.0
bleach==1.5.0
certifi==2018.8.24
chardet==3.0.4
click==6.7
cvxopt==1.2.1
cvxpy==0.2.24
cycler==0.10.0
decorator==4.3.0
dill==0.2.8.2
ecos==2.0.5
fastcache==1.0.2
Flask==1.0.2
Flask-JWT-Extended==3.12.1
Flask-RESTful==0.3.6
Flask-WTF==0.14.2
future==0.16.0
gast==0.2.0
geonamescache==1.0.1
grpcio==1.13.0
gunicorn==19.9.0
h5py==2.8.0
html5lib==0.9999999
idna==2.7
ipykernel==4.8.2
ipython==6.4.0
ipython-genutils==0.2.0
itsdangerous==0.24
jedi==0.12.1
Jinja2==2.10
jupyter-client==5.2.3
jupyter-core==4.4.0
Keras==2.2.0
Keras-Applications==1.0.2
Keras-Preprocessing==1.0.1
kiwisolver==1.0.1
Markdown==2.6.11
MarkupSafe==1.0
matplotlib==2.2.2
mongoengine==0.15.0
multiprocess==0.70.6.1
numpy==1.14.5
osqp==0.4.0
pandas==0.23.3
parso==0.3.0
patsy==0.5.0
pexpect==4.6.0
pickleshare==0.7.4
Pint==0.8.1
prompt-toolkit==1.0.15
protobuf==3.6.0
ptyprocess==0.6.0
Pygments==2.2.0
PyJWT==1.4.2
pymongo==3.7.0
pyparsing==2.2.0
python-dateutil==2.7.3
pytz==2018.5
PyYAML==3.13
pyzmq==17.0.0
requests==2.19.1
schedule==0.5.0
scikit-learn==0.19.1
scipy==1.1.0
scs==2.0.2
seaborn==0.8.1
simplegeneric==0.8.1
six==1.11.0
sklearn==0.0
statsmodels==0.9.0
talos==0.1.9
tensorboard==1.8.0
tensorflow==1.5.0
tensorflow-tensorboard==1.5.1
termcolor==1.1.0
toolz==0.9.0
tornado==5.0.2
traitlets==4.3.2
urllib3==1.23
wcwidth==0.1.7
Werkzeug==0.14.1
WTForms==2.2.1

**Comment by gabrieldemarmiesse** [comment]
can you do `pip install --upgrade keras` and tell me if the bug is still there?

**Comment by SpirosChadoulos** [comment]
I did it, bug is still here. If you search keras open issues you will find a similar issue.

**Comment by gabrieldemarmiesse** [comment]
I found something. So this issue has been fixed in #11133. You'll be able to see the changes in the next keras release.

**Comment by SpirosChadoulos** [comment]
Nice, thanks a lot!!
